scripts/migrations/2020-10-archivematica_bags_migration.py

```diff
@@ -0,0 +1,349 @@
+
+    with open(file_location) as fp:
+        for line in fp:
+            first, second = line.split(" ", 1)
+
+            if key_first:
+                fields[first.strip()] = second.strip()
+            else:
+                fields[second.strip()] = first.strip()
+
+    return fields
+
+
+class ArchivematicaUUIDBagMigrator:
+    def __init__(
+        self, workflow_s3_client, storage_s3_client, storage_client, s3_upload_bucket
+    ):
+        self.workflow_s3_client = workflow_s3_client
+        self.storage_s3_client = storage_s3_client
+        self.storage_client = storage_client
+        self.s3_upload_bucket = s3_upload_bucket
+
+        self.target_folder = "target"
+        self.tagmanifest_name = "tagmanifest-sha256.txt"
+        self.s3_upload_prefix = "born-digital/archivematica-uuid-update"
+
+    @staticmethod
+    def _get_archivematica_uuid(bucket, path, version):
+        files = filter_s3_objects(
+            s3_client=storage_s3_client,
+            bucket=bucket,
+            prefix=f"{path}/{version}/data/METS.",
+        )
+
+        assert len(files) == 1, files
+        mets_file_with_id = files[0]
+
+        archivematica_uuid = mets_file_with_id.split("/METS.")[-1].split(".xml")[0]
+
+        assert UUID(archivematica_uuid, version=4)
+        return archivematica_uuid
+
+    @staticmethod
+    def _generate_updated_checksums(working_folder):
+        files_in_need_of_update = ["bag-info.txt", "fetch.txt"]
+
+        return {
+            filename: generate_checksum(f"{working_folder}/{filename}")
+            for filename in files_in_need_of_update
+        }
+
+    def _load_existing_checksums(self, working_folder):
+        tag_manifest = load_space_separated_file(
+            file_location=f"{working_folder}/{self.tagmanifest_name}", key_first=False
+        )
+
+        files_that_should_be_referenced = [
+            "bag-info.txt",
+            "bagit.txt",
+            "manifest-sha256.txt",
+        ]
+
+        assert any(
+            filename in files_that_should_be_referenced
+            for filename in tag_manifest.keys()
+        )
+
+        return tag_manifest
+
+    @staticmethod
+    def _write_fetch_file(bucket, path, working_folder, files):
+        path_prefix = f"s3://{bucket}/{path}"
+
+        with open(f"{working_folder}/fetch.txt", "w") as fetch_file:
+            for file in files:
+                s3_uri = f"{path_prefix}/{file['path']}"
+                fetch_file.write(f"{s3_uri}\t{file['size']}\t{file['name']}\n")
+
+    @staticmethod
+    def _get_bagit_files_from_s3(
+        bucket, path, version, working_folder, tagmanifest_files
+    ):
+        file_locations = [
+            f"{path}/{version}/{file['name']}" for file in tagmanifest_files
+        ]
+
+        for location in file_locations:
+            filename = location.split("/")[-1]
+            save_path = f"{working_folder}/{filename}"
+            storage_s3_client.download_file(bucket, location, save_path)
```

Comment:
minor: prefer `os.path.basename(…)` instead of manipulating paths directly, for portability.

Also, you already have the filename as `file['name']`.